libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -72,15 +73,15 @@ object MapboxMetricsReporter : MetricsReporter {
     fun disable() {
         removeObserver()
         mapboxTelemetry.disable()
-        threadWorker.stop()
+        mainJobController.job.cancelChildren()
     }
 
     override fun addEvent(metricEvent: MetricEvent) {
         metricEvent.toTelemetryEvent()?.let {
             mapboxTelemetry.push(it)
         }
 
-        threadWorker.post {
+        mainJobController.scope.launch {
```

Comment:
why main thread is used? previous  `threadWorker` used `HandlerThread` to process work, so it was in background